internal/dag/conditions.go

```diff
@@ -30,13 +31,20 @@ func pathCondition(conds []projcontour.Condition) Condition {
 	}
 }
 
-func pathConditionsValid(conds []projcontour.Condition) bool {
-	found := 0
+// pathConditionsValid validates a slice of Conditions can be correctly merged.
+// It encodes the business rules about what is allowed for prefix Conditions.
+func pathConditionsValid(sw *ObjectStatusWriter, conds []projcontour.Condition, conditionsContext string) bool {
+	prefixCount := 0
 	for _, cond := range conds {
 		if cond.Prefix != "" {
-			found++
+			prefixCount++
+			if !strings.HasPrefix(cond.Prefix, "/") {
```

Comment:
suggestion: we know that cond.Prefix != "" at this point so you can just write
```
if cond.Prefix[0] != '/' {
```

Reply:
more correctly len(cond.Prefix) > 0 so its safe to access subscript 0

Reply:
of course. Neat.